Tests/BackendIntegrationTests/SubscriberAttributesManagerIntegrationTests.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  SubscriberAttributesManagerIntegrationTests.swift
+//
+//  Created by Nacho Soto on 4/1/22.
+
+@testable import RevenueCat
+
+import Nimble
+import XCTest
+
+// swiftlint:disable:next type_name
+class SubscriberAttributesManagerIntegrationTests: BaseBackendIntegrationTests {
+
+    func testNothingToSync() {
+        expect(Purchases.shared.syncSubscriberAttributesIfNeeded()) == 0
+    }
+
+    func testSyncOneAttribute() async throws {
+        Purchases.shared.setEmail("test@revenuecat.com")
+
+        let errors = await self.syncAttributes()
+        verifyAttributesSyncedWithNoErrors(errors, 1)
+    }
+
+    func testSettingTheSameAttributeDoesNotNeedToChangeIt() async throws {
+        let email = "test@revenuecat.com"
+        Purchases.shared.setEmail(email)
+
+        var errors = await self.syncAttributes()
+        verifyAttributesSyncedWithNoErrors(errors, 1)
+
+        Purchases.shared.setEmail(email)
+        errors = await self.syncAttributes()
+        verifyAttributesSyncedWithNoErrors(errors, 0)
+    }
+
+    func testChangingEmailSyncsIt() async throws {
+        Purchases.shared.setEmail("test@revenuecat.com")
+
+        var errors = await self.syncAttributes()
+        verifyAttributesSyncedWithNoErrors(errors, 1)
+
+        Purchases.shared.setEmail("test2@revenuecat.com")
+        errors = await self.syncAttributes()
+        verifyAttributesSyncedWithNoErrors(errors, 1)
+    }
+
+    func testSyncInvalidEmail() async throws {
+        Purchases.shared.setEmail("invalid @ email @.com")
+
+        let errors = await self.syncAttributes()
+        expect(errors).to(haveCount(1))
+
+        let error = try XCTUnwrap(errors.first ?? nil) as NSError
+        expect(error.domain) == RCPurchasesErrorCodeDomain
+        expect(error.code) == ErrorCode.invalidSubscriberAttributesError.rawValue
+        expect(error.subscriberAttributesErrors) == [
+            "$email": "Email address is not a valid email."
+        ]
```

Comment:
This is a failing test that should be fixed by #1429.

Reply:
Confirmed, this test will pass once `ErrorResponse` fixes attribute error parsing.